cdap-docs/reference-manual/source/http-restful-api/http-restful-api-v3/preferences.rst

```diff
@@ -0,0 +1,163 @@
+.. meta::
+:author: Cask Data, Inc.
+    :description: HTTP RESTful Interface to the Cask Data Application Platform
+      :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _http-restful-api-preferences:
+.. _http-restful-api-v3-namespace:
+
+===========================================================
+Preferences HTTP RESTful API
+===========================================================
+
+.. highlight:: console
+
+Use the CDAP Preferences HTTP API to save, retrieve, delete preferences in CDAP.
+
+Preferences, their use and examples, are described in the :ref:`Admin' Manual: Preferences
+  <preferences>`.
+
+  For the remainder of this API, it is assumed that the preferences you are using is defined
+  by the ``<base-url>``, as descibed under :ref:`Conventions <http-restful-api-conventions>`.
+
+Set Preferences
+------------------
+To set preferences for the CDAP Instance, Namespace, Application, Program submit an HTTP PUT request::
+
+  PUT http://<host>:<port>/v3/configuration/preferences/
+
+  PUT http://<host>:<port>/v3/configuration/preferences/namespaces/<namespace-id>
+
+  PUT http://<host>:<port>/v3/configuration/preferences/namespaces/<namespace-id>/apps/<app-id>
+
+  PUT http://<host>:<port>/v3/configuration/preferences/namespaces/<namespace-id>/apps/<app-id>/<program-type>/<program-id>
+
+.. list-table::
+:widths: 20 80
+   :header-rows: 1
+
+     * - Parameter
+       - Description
+     * - ``<namespace-id>``
+     - Namespace ID
+     * - ``<app-id>``
+     - Application ID
+     * - ``<program-type>``
+     - Program Type (flows, services, etc)
+     * - ``<program-id>``
+     - Program ID
+
+Preferences can be set only for entities that exist. For example, Preferences cannot be set for a Namespace
+that does not exist or an application that has not be deployed. Properties, which are map of string-string pairs are
```

Comment:
Properties, as a map of string-string pairs, are passed in the JSON request body.

Preferences can be set only for entities that exist. For example, Preferences cannot be set for a Namespace
that does not exist or an application that has not yet been deployed.